### ISSUE TITLE
Bug: [explicit-function-return-type] False positive for methods inside a returned object

### ISSUE THREAD

**Issue by hgl** [typescript-eslint/typescript-eslint#12090]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=5.9.2&fileType=.tsx&code=JYOwLgpgTgZghgYwgAgGIHt3IN4ChnIBGcUAFAJQBcyAzmFKAOa4C%2BuuMAriAmMOiGQANCtQxY8BKBDCcogyQSIkKOfEqky5ggERwd6gmyOsgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6RAD2MbMroDNYTMvkoB7JrWiJ8saJKKl0URNGhjokcGAC%2BIHUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface Foo {
  bar(): string
}

function X(): Foo {
  return {
    bar() {
      return "a"
    }
  }
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/explicit-function-return-type": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error.

### Actual Result

Missing return type on function. 7:5 - 7:8

**Comment by bradzacher** [comment]
Why do you believe this is a false positive? You didn't include any information in your issue.

**Comment by hgl** [comment]
Sorry, let me elaborate.

`bar()`'s return type should already be determinable from contextual typing. Explicitly specifying a return type here is usually not a good idea IMHO, since it's unnecessary and you have to make sure the return type matches the contextual type.

I can understand if you think this rule by default should always require explicit return type in this case, but would you be open to supporting an option that allow omitting return types where they are already provided from contextual typing?

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/8918
